src/collection-view.js

```diff
@@ -128,12 +128,33 @@ Marionette.CollectionView = Marionette.View.extend({
     return this;
   },
 
+  // Reorder DOM after sorting. When your element's rendering
+  // do not use their index, you can pass reorderOnSort: true
+  // to only reorder the DOM after a sort instead of rendering
+  // all the collectionView
+  reorder: function () {
+    var children = this.children;
+
+    // get the DOM nodes in the same order as the models
+    var els = this.collection.models.map(function (model) {
```

Comment:
hmm not sure if it would be slower or not.  above is a more functional approach ^ that required less internal state. 
but not a real big deal one way or the other